micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java

```diff
@@ -208,8 +219,27 @@ private void countPoolSizeDelta(Map<String, MemoryUsage> before, Map<String, Mem
         }
     }
 
-    private boolean isMajorGc(String gcName) {
-        return !isGenerationalGc || GcGenerationAge.fromGcName(gcName) == GcGenerationAge.OLD;
+        private boolean shouldUpdateDataSizeMetrics(String gcName) {
+            return nonGenerationalGcShouldUpdateDataSize(gcName) || isMajorGenerationalGc(gcName);
+        }
+
+        private boolean isMajorGenerationalGc(String gcName) {
+            return GcGenerationAge.fromGcName(gcName) == GcGenerationAge.OLD;
+        }
+
+        private boolean nonGenerationalGcShouldUpdateDataSize(String gcName) {
+            return !isGenerationalGc
+                    // Skip Shenandoah and ZGC gc notifications with the name Pauses due to missing memory pool size info
+                    && !gcName.endsWith("Pauses");
```

Comment:
Hard to see in the diff (sorry for refactoring while bug fixing), but this is the key change.

Reply:
Though as I'm thinking about it more, I'm not sure why we even need to consider non-generational collectors like this at all. I don't think there will be a GC notification for a non-generational collector where the `longLivedAfter < longLivedBefore` condition won't be met.